cynthion/python/Makefile

```diff
@@ -62,3 +82,53 @@ assets/moondancer.bin:
 			python -m cynthion.gateware.$(basename $@).top --dry-run --output "assets/$$lp/$@"; \
 		fi; \
 	done
+
+
+# - environment checks --------------------------------------------------------
+
+detect-fpga-toolchain:
+ifeq (, $(shell which nextpnr-ecp5))
```

Comment:
Ugh, okay so two things:

1. GNU [Make conditionals](https://www.gnu.org/software/make/manual/html_node/Conditional-Example.html) are not indented
2. That's really weird, doesn't happen for me -- tested on macOS GNU Make 3.81 and Arch Linux GNU Make 4.4.1.

What do you mean by running every time? Can you please show me your command line & output?

Reply:
Oh *duh* please don't mind me. GNU Make conditionals are always evaluated during the first pass which meant that if you did not have a Cynthion plugged in you'd be getting Apollo errors.

I've replaced most of the Make conditionals with shell conditionals to defuse the footgun and also caught another issue where if Cynthion wasn't plugged in it would tell you that you need to upgrade it to the latest version: 1659ad0240501824a0a06e8df902049897b74126

Reply:
Hmm. I'm not sure why indenting the conditionals appear to work for me! Anyway, I like your solution.

Reply:
What is the purpose of `|| [ $$? -eq 0 ]`? It seems redundant to me.

Reply:
> Hmm. I'm not sure why indenting the conditionals appeared to work for me! Anyway, I like your solution.

Because they don't get called then.

Reply:
> What is the purpose of `|| [ $$? -eq 0 ]`? It seems redundant to me.

Semantic equivalence with the code they're replacing. i.e. ability to check specific return codes. Explicit is better than implicit.